Abyss.Web/ClientApp/src/app/servermanager/servermanager.component.ts

```diff
@@ -7,21 +8,69 @@ import { DialogService } from '../services';
 
 @Component({
     templateUrl: './servermanager.component.html',
+    styleUrls: ['./servermanager.component.scss'],
 })
 export class ServerManagerComponent implements OnInit {
     public servers?: IServer[];
     public ServerStatus = ServerStatus;
     public selectedId?: string;
     public loading = false;
+    public log: string[] = [];
+    public hubReady = false;
+    private hub = new HubConnectionBuilder()
```

Comment:
I'd personally move all of the hub creations and subscriptions out to it's own `HubManagerService.ts` that exposes what the components need. 

You can then in the `app.component.ts` just fire up the connections required and use the service to communicate any events needed using EventEmitters

Reply:
Yeah not a bad shout, there is a fair bit of copy pasted code